zaza/openstack/charm_tests/ceph/mon/tests.py

```diff
@@ -0,0 +1,200 @@
+        # Skip this test if release is less than xenial_ocata as in that case
+        # cinder HAS a relation with ceph directly and this test would fail
+        current_release = openstack_utils.get_os_release()
+        xenial_ocata = openstack_utils.get_os_release('xenial_ocata')
+        if current_release < xenial_ocata:
+            logging.info("Skipping test as release < xenial-ocata")
+            return
+
+        units = zaza.model.get_units("cinder-ceph",
+                                     model_name=self.model_name)
+
+        # check each unit prior to breaking relation
+        for unit in units:
+            dir_list = directory_listing(unit.name, "/etc/ceph")
+            if 'ceph.conf' in dir_list:
+                logging.debug(
+                    "/etc/ceph/ceph.conf exists BEFORE relation-broken")
+            else:
+                raise zaza_exceptions.CephGenericError(
+                    "unit: {} - /etc/ceph/ceph.conf does not exist "
+                    "BEFORE relation-broken".format(unit.name))
+
+        # remove the relation so that /etc/ceph/ceph.conf is removed
+        logging.info("Removing ceph-mon:client <-> cinder-ceph:ceph relation")
+        zaza.model.remove_relation(
+            "ceph-mon", "ceph-mon:client", "cinder-ceph:ceph")
+        # zaza.model.wait_for_agent_status()
+        logging.info("Wait till relation is removed...")
+        ceph_mon_units = zaza.model.get_units("ceph-mon",
+                                              model_name=self.model_name)
+        conditions = [
+            invert_condition(
+                does_relation_exist(
+                    u.name, "ceph-mon", "cinder-ceph", "ceph",
+                    self.model_name))
+            for u in ceph_mon_units]
+        zaza.model.block_until(*conditions)
+
+        logging.info("Checking each unit after breaking relation...")
+        for unit in units:
+            dir_list = directory_listing(unit.name, "/etc/ceph")
+            if 'ceph.conf' not in dir_list:
+                logging.debug(
+                    "/etc/ceph/ceph.conf removed AFTER relation-broken")
+            else:
+                raise zaza_exceptions.CephGenericError(
+                    "unit: {} - /etc/ceph/ceph.conf still exists "
+                    "AFTER relation-broken".format(unit.name))
+
+        # Restore cinder-ceph and ceph-mon relation to keep tests idempotent
+        logging.info("Restoring ceph-mon:client <-> cinder-ceph:ceph relation")
+        zaza.model.add_relation(
+            "ceph-mon", "ceph-mon:client", "cinder-ceph:ceph")
+        conditions = [
+            does_relation_exist(
+                u.name, "ceph-mon", "cinder-ceph", "ceph", self.model_name)
+            for u in ceph_mon_units]
+        logging.info("Wait till model is idle ...")
+        zaza.model.block_until(*conditions)
+        zaza.model.block_until_all_units_idle()
+        logging.info("... Done.")
+
+
+def does_relation_exist(unit_name,
```

Comment:
This should probably be a more generic helper in Zaza. I'm tentatively OK not blocking on this so we can get it landed more quickly, but I'd like to see a bug to track moving this into Zaza

Reply:
Yes, I'll raise a bug.

Reply:
Thanks!